### ISSUE TITLE
pyright reports erroneous errors when union types are used in isinstance calls

### ISSUE THREAD

**Issue by thisisrandy** [microsoft/pyright#2636]
**Describe the bug**
[PEP 604 allows for union types](https://www.python.org/dev/peps/pep-0604/#isinstance-and-issubclass) in `isinstance` calls. However, `pyright` complains that such calls are not valid.

**To Reproduce**
Run the below code through `pyright` and observe the error.

**Expected behavior**
`pyright` should not complain.

**Screenshots or Code**
From PEP 604:

```python
isinstance("", int | str)
```
`pyright` output:

>error: Argument of type "Type[int] | Type[str]" cannot be assigned to parameter "__class_or_tuple" of type "type | UnionType | Tuple[type | UnionType | Tuple[Any, ...], ...]" in function "isinstance"
>    Type "object" cannot be assigned to type "type | UnionType | Tuple[type | UnionType | Tuple[Any, ...], ...]"
>      "object" is incompatible with "type"
>      "object" is incompatible with "UnionType"
>      "object" is incompatible with "Tuple[type | UnionType | Tuple[Any, ...], ...]" (reportGeneralTypeIssues)

**VS Code extension or command-line**
```
> pyright --version   
pyright 1.1.186
```

**Comment by erictraut** [comment]
You're running an old version of pyright. The latest version is 1.1.190. This issue was fixed several versions ago.

**Comment by thisisrandy** [comment]
Ah, my apologies. 1.1.190 does indeed have the correct behavior.

I'm still seeing the error in VS Code on pylance v2021.11.2. Can I assume that the upcoming release (Wednesdays, you said elsewhere) will update to the latest version of pyright?

**Comment by bschnurr** [comment]
Sorry, last week we didn't have a pylance release. this wednesday we plan to.

**Comment by thisisrandy** [comment]
Wonderful, thanks. I look forward to it.
